src/layerelement.cpp

```diff
@@ -2517,6 +2545,14 @@ int LayerElement::LayerElementsInTimeSpan(FunctorParams *functorParams) const
     // The element is starting after the event end - we can stop here
     if (time >= (params->m_time + params->m_duration)) return FUNCTOR_STOP;
 
+    if (this->Is(NOTE)) {
+        const Note *note = vrv_cast<const Note *>(this);
+        if (note->IsChordTone()
+            && (std::find(params->m_elements.begin(), params->m_elements.end(), this->GetParent())
```

Comment:
You also need to look for the chord (which is not always the parent)
```suggestion
        const Chord *chord = note->IsChordTone();
        if (chord
            && (std::find(params->m_elements.begin(), params->m_elements.end(), chord)
```